cherryontech/comet-cruisers-spring2024
# Update/todo list sav&css

## Jira Story Card / Issue Link
[Unsaved Changes To-Do]https://cometcruisers.atlassian.net/browse/SCRUM-41

## Description of PR:
- Changed todo list save to auto save by persisting the data
- - removed the save button
- add titles to icons and todo-list/task names fore readability
- edit tutorial to be more accurate about drag and drop
- simplify code to be more readable
- changed html nesting so that the button is inside the link: this allows the whole button to be clickable
- changed popup code so it does not show up if there were no changes
- - this was done by having a useState for hasChanges shared between the JournalContainer component and the JournalTextEntry and have a ternary statement for the html around the go back arrow
## Type of change in this PR:

## Testing instructions:
Check if the popup shows up when you make changes and make sure it doesn't if you didn't for both old and new entries
## What I learned/had trouble with:
n/a
## Checklist before requesting a review
- [x] I have refactored my code before pushing
- [x] My commit message follows standard guidelines (eg. “If applied, my commit will (INSERT COMMIT MESSAGE HERE)”)
